communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java

```diff
@@ -39,13 +49,43 @@
 @ApplicationScoped
 public class AproxConnectorImpl implements AproxConnector {
 
-    private final Configuration config = new Configuration();
+    @Inject
+    private Configuration config;
+
+    @Inject
+    private SCM scmManager;
+
+    @Inject
+    private PomAnalyzer pomAnalyzer;
 
     @Override
-    public Optional<GAVDependencyTree> getDependencyTreeOfRevision(String scmUrl, String revision,
-            String pomPath) {
-        // TODO Auto-generated method stub
-        return Optional.empty();
+    public Optional<List<ScmGAVRelationship>> getDependencyTreeOfRevision(String scmUrl,
+            String revision, String pomPath) throws ScmException, PomAnalysisException {
+
+        try {
+            // git clone
+            // TODO: hardcoded to git right now
```

Comment:
I'd say it is not a problem to have support only for GIT for version 0.4.0. Feel free to implement it only for GIT and create issue(s) for it.